libvast/src/segment_store.cpp

```diff
@@ -427,9 +428,10 @@ caf::expected<segment> segment_store::load_segment(uuid id) const {
              filename);
   auto chk = chunk::mmap(filename);
   if (!chk)
-    return caf::make_error(ec::filesystem_error, "failed to mmap chunk",
-                           filename.string());
-  if (auto segment = segment::make(std::move(chk))) {
+    return caf::make_error(ec::filesystem_error,
+                           fmt::format("failed to mmap chunk {}: {}", filename,
+                                       caf::to_string(chk.error().context())));
```

Comment:
Same as above: It suffices to forward the error here.

```suggestion
    return std::move(chk.error());
```